### PR TITLE
handle malformed id's when describing snapshots

### PR DESCRIPTION
**Pull request by thisisshi**
When running policies with `not-encrypted` filter, malformed snapshot ID's can prevent policy executions to complete, this PR handles malformed ID's similarly to how it handles snapshots that are not found.

### CODE REVIEW

**Comment by kapilt** [comment]
ideally with a test around the bad snap case, via just passing an exception to the method/function, but else lgtm, thanks.

File: c7n/resources/asg.py [hunk 1]
```diff
@@ -526,21 +526,32 @@ def get_unencrypted_configs(self, ec2):
...
-                    snap_ids.remove(e_snap_id)
+                bad_snap = self.get_bad_snapshot(e)
+                if bad_snap:
+                    snap_ids.remove(bad_snap)
                     continue
                 raise
             else:
                 return result.get('Snapshots', ())
 
+    def get_bad_snapshot(self, e):
```
Comment (kapilt) [inline c7n/resources/asg.py]: this could be a static function which should also make testing easier.

Comment (thisisshi) [reply c7n/resources/asg.py]: Yeah, I was thinking of moving it into utils

File: c7n/utils.py [hunk 2]
```diff
@@ -489,3 +489,18 @@ def format_string_values(obj, *args, **kwargs):
         return obj.format(*args, **kwargs)
     else:
         return obj
+
+def get_bad_snapshot(e):
```
Comment (kapilt) [inline c7n/utils.py]: actually it was fine to just leave this on the class and use a `staticmethod` decorator to ease the testing, if we want to share it between multiple callers its probably best to move it to the ebs module, as its not really a general purpose utility its pretty specific to the ebs resource.

Comment (thisisshi) [reply c7n/utils.py]: makes sense-- I was thinking we would want to abstract out this type of string parsing to centralize changes if the messages themselves change the way that they're formatted, perhaps a more generic method like `get_bad_resource` to deal with this type of try/catch behavior

Comment (kapilt) [reply c7n/utils.py]: yeah i think we'd considered something like that before, but there really isn't any good way to handle these generically, its error message parsing, and it varies on message content by api, and potentially also varies over time. if we can get this stuff out of the error handlers and into a set of functions that we can test thats a win.

**Pull request merged**